registry/api/protocol/eureka/instance_handler.go

```diff
@@ -183,9 +166,18 @@ func (routes *Routes) registerInstance(w rest.ResponseWriter, r *rest.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
-func validateJSON(jsonString json.RawMessage) bool {
-	var js interface{}
-	return json.Unmarshal(jsonString, &js) == nil
+func extractTagsFromMetadata(jsonString json.RawMessage) ([]string, error) {
+	var md map[string]interface{}
+	if err := json.Unmarshal(jsonString, &md); err != nil {
+		return nil, err
+	}
+
+	var tags []string
+	if val, ok := md[metadataTags]; ok {
+		sval := fmt.Sprintf("%v", val)
+		tags = strings.Split(sval, metadataTagsDelimiter)
```

Comment:
`tags = strings.Split(sval, metadataTagsDelimiter)`
I think we need to trim whitespaces around the split tags, so that users will be able to specify them as `tag1, tag2`.